### ISSUE TITLE
Unable to create 2 cards for 2 devices zhimi.humidifier.ca4

### ISSUE THREAD

**Issue by diemon24** [artem-sedykh/mini-humidifier#98]
have a very strange card behavior. I have 2 identical humidifiers via Xiaomi Miio integration (https://www.home-assistant.io/integrations/xiaomi_miio). I connect 1 humidifier and get the states of all sensors and switches. But on the second humidifier I don't get this card configuration. About the humidifier work when you turn on the mini-humidifier card. The sensors of both devices are controlled from a custom card.

`type: custom:mini-humidifier
entity: humidifier.zhimi_humidifier_ca4_2
model: zhimi.humidifier.ca4
toggle:
  default: 'on'
  hide: 'on'
`
<img width="750" alt="Снимок экрана 2022-07-09 в 16 15 15" src="https://user-images.githubusercontent.com/91910101/178107382-be5b8ee3-dc26-4c6f-8e8c-fc7d553c3571.png">
<img width="750" alt="Снимок экрана 2022-07-09 в 16 15 30" src="https://user-images.githubusercontent.com/91910101/178107384-aca78105-b9cb-483a-b144-eeba6d880c50.png">

HUCS: 1.25.5
Home Assistant Core 2022.6.6

**Comment by regevbr** [comment]
if you have just a single card with the entity that doers work, it suddenly works?

**Comment by regevbr** [comment]
the model expects to have sensors in the format of `sensor.{entity_id}_{sensor}` are you sure you have those for the 2nd humidifier?

**Comment by diemon24** [comment]
> the model expects to have sensors in the format of `sensor.{entity_id}_{sensor}` are you sure you have those for the 2nd humidifier?

Yes. I have all working sensors.

**Comment by diemon24** [comment]
> if you have just a single card with the entity that doers work, it suddenly works?

In custom card all sensors and switch working. Only mini-card humidifier.zhimi_humidifier_ca4_2 not working.

**Comment by regevbr** [comment]
sorry but none of that answers my questions... Do you have an entity with id humidifier.zhimi_humidifier_ca4_2_motor_speed?

**Comment by diemon24** [comment]
> sorry but none of that answers my questions... Do you have an entity with id humidifier.zhimi_humidifier_ca4_2_motor_speed?

yes. I have the same set of sensors and switches for two humidifiers. In the message above, i attached the entire list of sensors and switches for the humidifier, which is not displayed well on the card mini-humidufier. 

This strange behavior and makes me smile. Because I have two identical sets of sensors and switches for two humidifiers. All this works in a standard card, but does not work in mini-humidufier card.

**Comment by regevbr** [comment]
what you posted is a list of sensor names, not sensor entity ids...
So if you delete the card that works and just have a single card with the _2 entity, it works?

**Comment by diemon24** [comment]
> what you posted is a list of sensor names, not sensor entity ids...
> So if you delete the card that works and just have a single card with the _2 entity, it works?

 If I delete work card and create card for _2 entity, is not work anyway

**Comment by regevbr** [comment]
Good, we are getting somewhere. Now please send me the **entity ids** of all the sensors of the nonworking humidifier

**Comment by diemon24** [comment]
> ood, we are getting somewhere. Now please send me the **entity ids** of all the sensors of the nonworking humidifier

sensor.zhimi_humidifier_ca4_humidity_2
sensor.zhimi_humidifier_ca4_temperature_2
switch.zhimi_humidifier_ca4_buzzer_2
switch.zhimi_humidifier_ca4_child_lock_2
switch.zhimi_humidifier_ca4_clean_mode_2
switch.zhimi_humidifier_ca4_dry_mode_2
number.zhimi_humidifier_ca4_motor_speed_2
sensor.zhimi_humidifier_ca4_actual_speed_2
sensor.zhimi_humidifier_ca4_water_level_2
binary_sensor.zhimi_humidifier_ca4_water_tank_2

**Comment by regevbr** [comment]
Great so that is the reason, please refer to my first comment on how entity ids should be for the card to work

**Comment by diemon24** [comment]
> Great so that is the reason, please refer to my first comment on how entity ids should be for the card to work

yes, but Xiaomi Miio does not allow changing the device name when adding...

**Comment by regevbr** [comment]
You can change the entity ids yourself i believe

**Comment by diemon24** [comment]
> You can change the entity ids yourself i believe

Ok. I changed all sensors to name sensor.{entity_id}2_{sensor} and create a new card. This working. Thx you very much. I suggest updating this issue.

**Comment by artem-sedykh** [comment]
Closing this, with the conclusion written down where the next person will find
it - and with the card doing better than it did for you.

**What it was.** The bundled models build the ids of the entities they read out
of the humidifier's own: `sensor.<entity>_humidity`,
`switch.<entity>_child_lock`, and so on. Home Assistant appends `_2` at the
**end** for a second device of the same kind, so your entities came out as
`sensor.zhimi_humidifier_ca4_humidity_2` while the preset looked for
`sensor.zhimi_humidifier_ca4_2_humidity`. Every control whose entity was not
found was skipped, which is why the second card rendered nearly empty.

**What was wrong on our side** is that this took fourteen comments to establish.
The card said nothing at all: a control with a missing entity was left out
silently, on the dashboard and in the console alike. From the next release it
names it:

```
mini-humidifier: indicator 'humidity' reads sensor.zhimi_humidifier_ca4_2_humidity,
which does not exist in Home Assistant, so it is not shown. Check the entity id -
the bundled models build it from the humidifier's own.
```

Renaming the entity ids, as you did, is still the tidiest fix. The alternative,
if renaming is not an option, is to point the indicators and buttons at the real
ids with `source: entity` and `state: entity` - `docs/indicators.md` and
`docs/buttons.md` show both.

Thank you for staying with it through that thread; the diagnostics came out of
re-reading it.
